SmartDeviceLink/private/SDLMenuManager.m

```diff
@@ -530,6 +529,62 @@ - (BOOL)sdl_isDynamicMenuUpdateActive:(SDLDynamicMenuUpdatesMode)dynamicMenuUpda
     }
 }
 
+/// Checks if 2 or more cells have the same text/title. In case this condition is true, this function will handle the presented issue by adding "(count)".
+/// E.g. Choices param contains 2 cells with text/title "Address" will be handled by updating the uniqueText/uniqueTitle of the second cell to "Address (2)".
+/// @param choices The choices to be uploaded.
+- (void)sdl_addUniqueNamesToCells:(nullable NSArray<SDLMenuCell *> *)choices {
+    // Tracks how many of each cell primary text there are so that we can append numbers to make each unique as necessary
+    NSMutableDictionary<NSString *, NSNumber *> *dictCounter = [[NSMutableDictionary alloc] init];
+    for (SDLMenuCell *cell in choices) {
+        NSString *cellName = cell.title;
+        NSNumber *counter = dictCounter[cellName];
+        if (counter != nil) {
+            counter = @(counter.intValue + 1);
+            dictCounter[cellName] = counter;
+        } else {
+            dictCounter[cellName] = @1;
+        }
+
+        counter = dictCounter[cellName];
+        if (counter.intValue > 1) {
+            cell.uniqueTitle = [NSString stringWithFormat: @"%@ (%d)", cell.title, counter.intValue];
+        }
+
+        if (cell.subCells.count > 0) {
+            [self sdl_addUniqueNamesToCells:cell.subCells];
+        }
+    }
+}
+
+/**
+ Check for cell lists with completely duplicate information, or any duplicate voiceCommands
+
+ @param cells The cells you will be adding
+ @return Boolean that indicates whether menuCells are unique or not
+ */
+-(BOOL)sdl_menuCellsAreUnique:(NSArray<SDLMenuCell *> *)cells {
+    ///Check all voice commands for identical items and check each list of cells for identical cells
+    NSMutableSet *identicalCellsCheckSet = [NSMutableSet set];
+    for (SDLMenuCell *cell in cells) {
+        [identicalCellsCheckSet addObject:cell];
+        if (cell.subCells.count > 0 && ![self sdl_menuCellsAreUnique:cell.subCells]) {
+            return NO;
+        }
+
+        if (cell.voiceCommands == nil) { continue; }
+        [identicalVoiceCommandsCheckSet addObjectsFromArray:cell.voiceCommands];
+        voiceCommandCount += cell.voiceCommands.count;
```

Comment:
This probably should be broken out. It's not a part of the check that the method name describes: `menuCellsAreUnique`, they can be unique and still have duplicate voice commands.